### ISSUE TITLE
Changing VLAN-Tags of remote devices via librenms

### ISSUE THREAD

**Issue by f0o** [librenms/librenms#71]
As discussed on IRC and within Roadmap.

This issue report is meant to be a pool of features this enhancement should have as well as structural discussion.
I will write the code as demanded by comments or IRC.

```text
08:26:24 blahdeblah | I think we'd be silly to try to support everything. We should make it work well enough to meet the needs of those who want the feature, but trying
                    | to make it suit people who aren't even using it isn't necessary, IMO.
08:27:34 blahdeblah | e.g. My work networks use HP & Cisco - I think both of them have a read-write SNMP implementation, so if it works with them, I'm happy.  Then if we
                    | put the feature out there and other people try it and want it to work for them, they can submit patches.
08:28:37        f0o | true true
08:29:21        f0o | well I might give it a shot, at least the WebUI and a switching php can be done without even looking at the snmp implementations of the hardware
08:29:44        f0o | gonna keep it simple so the snmp instructions for given hardware can be added later without a big fuzz of code
08:29:51 blahdeblah | yep - a front end that we could add different backends to would be great
09:03:26    lafwood | When you're doing it it would be a good idea to validate what vlans exist on the switch before changing one on a port.
09:03:48    lafwood | I know on cisco if you change the ports vlan to one that doesn't exist then that vlan is created.
09:04:04    lafwood | Bad times may be had.
09:17:16 blahdeblah | Indeed
09:17:49 blahdeblah | There would need to be a set of primitives implemented, e.g. check for VLAN existence, check for port existence, etc.
09:18:07 blahdeblah | The more I think about it, the bigger a nightmare it becomes ;-\
09:18:39 blahdeblah | We'd basically have to reinvent SNMP at the API level, and then some...
09:20:28    lafwood | :) Certainly wouldn't be simple
```

PS: Can a mod/admin change the status to 'Enhancement' and add me as "assigned to"? Thanks

**Comment by Lonecrowe** [comment]
What f0o said - snmpv3  
"Transport Layer Security is the next generation of Secure Sockets Layer (SSL) and has been added to the SNMPv3 architecture."

So any of the devices out there that support V3 will be pretty much on top of their snmp game and will have for the most part well written mibs.  

Allowing use of V2c within IP space outside of private ranges could be tagged with a disclaimer box someone would need to check to override it and assume liability.  

Seeing as how that snmpset will be sent probably only once in a blue moon and not at regular intervals I think its going to be difficult to intercept.  The PDU idea is just one example of rebooting one of the ports etc.  Another example is sometimes we need to disable the ethernet port on a radio to force the customer router to power down and then reacquire dhcp.

**Comment by killsudo** [comment]
You might check out http://www.rconfig.com/about-rconfig as its written in PHP and has the ability the perform configuration compliance management via subsections in a config like comparing acl's across two or more devices to ensure the acl is the same.

They also have bulk device config change support as discussed here.

I'm not sure if they have an api or if the developers would be interested in any cross-support integration with librenms.

**Comment by laf** [comment]
Proposing we close this and consider it out of scope for now.

**Comment by Lonecrowe** [comment]
Sure give up already!  :)

Yeah its a "maybe later" I think.

**Comment by laf** [comment]
Had a few :+1: for closing on irc. Can be re-opened at a later stage.
